compiler/resolution/preFold.cpp

```diff
@@ -390,7 +390,19 @@ static void setRecordDefaultValueFlags(AggregateType* at) {
       ts->addFlag(FLAG_TYPE_NO_DEFAULT_VALUE);
     } else if (isNilableClassType(at)) {
       ts->addFlag(FLAG_TYPE_DEFAULT_VALUE);
-
+    } else if (at->symbol->hasFlag(FLAG_TUPLE)) {
+      Flag flag = FLAG_TYPE_DEFAULT_VALUE;
+      for_fields(field, at) {
+        if (field->isRef()) continue;
```

Comment:
This seems odd to me. Shouldn't a tuple type containing a ref-to-owned C, say, indicate it is not default initialize-able?

Reply:
I'm confused, how do you default initialize a ref to anything? I guess I could grab the value type from the ref and mark that?

Reply:
I guess you should probably mark value types in a tuple even if it is referential, just to be safe?

Reply:
I think if it you tried to declare a variable with it, the compiler would convert it to a value tuple

Reply:
My thought was that the compiler will instantiate a different tuple type for the value tuple version of the tuple and it would be checked when it is default initialized.

However it probably doesn't hurt to be safe regardless.

Reply:
After thinking about it more, the compiler will probably (?) only ever default initialize value tuples...so they're all value types... 🤦